controllers/scenario/scenario_adapter.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/redhat-appstudio/integration-service/gitops"
+	"github.com/redhat-appstudio/operator-goodies/reconciler"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Adapter holds the objects needed to reconcile a Release.
+type Adapter struct {
+	application *applicationapiv1alpha1.Application
+	scenario    *v1alpha1.IntegrationTestScenario
+	logger      logr.Logger
+	client      client.Client
+	context     context.Context
+}
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(application *applicationapiv1alpha1.Application, scenario *v1alpha1.IntegrationTestScenario, logger logr.Logger, client client.Client,
+	context context.Context) *Adapter {
+	return &Adapter{
+		application: application,
+		scenario:    scenario,
+		logger:      logger,
+		client:      client,
+		context:     context,
+	}
+}
+
+// EnsureCreatedScenarioIsValid is an operation that ensures created IntegrationTestScenario is valid
+// in case it is, set its owner reference
+func (a *Adapter) EnsureCreatedScenarioIsValid() (reconciler.OperationResult, error) {
+
+	// First check if application exists or not
+	if a.application == nil {
+		a.logger.Info("Application for scenario was not found.",
+			"Sceario.Name", a.scenario.Name)
+
+		patch := client.MergeFrom(a.scenario.DeepCopy())
+		SetScenarioIntegrationStatusAsInvalid(a.scenario, "Failed to get application for scenario.")
+		err := a.client.Status().Patch(a.context, a.scenario, patch)
+		if err != nil {
+			a.logger.Error(err, "Failed to update Scenario",
+				"Scenario.Name", a.scenario.Name)
+			return reconciler.RequeueWithError(err)
+		}
+		return reconciler.ContinueProcessing()
+	}
+	// Checks if scenario has ownerReference assigned to it
+	if a.scenario.OwnerReferences == nil {
+		patch := client.MergeFrom(a.scenario.DeepCopy())
+		err := ctrl.SetControllerReference(a.application, a.scenario, a.client.Scheme())
+		if err != nil {
+			a.logger.Error(err, "Error setting owner reference.",
+				"Application.Name", a.application.Name,
+				"Application.Namespace", a.application.Namespace)
+			patch := client.MergeFrom(a.scenario.DeepCopy())
+			SetScenarioIntegrationStatusAsInvalid(a.scenario, "Failed to get application for scenario.")
+			err = a.client.Status().Patch(a.context, a.scenario, patch)
+			if err != nil {
+				a.logger.Error(err, "Failed to update Scenario",
+					"Scenario.Name", a.scenario.Name)
+				return reconciler.RequeueWithError(err)
+			}
+
+		}
+		err = a.client.Patch(a.context, a.scenario, patch)
+		if err != nil {
+			a.logger.Error(err, "Failed to update Scenario",
+				"Scenario.Name", a.scenario.Name)
+			return reconciler.RequeueWithError(err)
+		}
+
+	}
+	// Checks if scenario has environment defined
+	if reflect.ValueOf(a.scenario.Spec.Environment).IsZero() {
+		a.logger.Info("IntegrationTestScenario has no environment defined",
+			"Scenario.Name:", a.scenario.Name,
+			"Scenario.Namespace", a.scenario.Namespace)
+		return reconciler.ContinueProcessing()
```

Comment:
I think that this `ContinueProcessing` call doesn't need to happen since we still want to mark the Snapshot as valid later in the function even if the Environment isn't defined.

Reply:
Agreed.